src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -455,6 +468,22 @@ private async Task<bool> ReceiveHandshakeResponseAsync(PipeReader input, Cancell
             }
         }
 
+        private async Task UpdateAzureIdentityAsync(AccessKey key, TimerAwaitable timer)
+        {
+            if (key is AadAccessKey aadKey)
+            {
+                using (timer)
+                {
+                    timer.Start();
+                    while (await timer)
+                    {
+                        var message = new AccessKeyRequestMessage(await aadKey.GenerateAadTokenAsync());
```

Comment:
can `await aadKey.GenerateAadTokenAsync()` throw?

Reply:
The server cannot start if it throwed.